### PR TITLE
Update minerfund addresses for numbers upgrade

### PR DESCRIPTION
**Pull request by schancel**
Use new minerfund addresses for numbers upgrade and change the way they're handled so that they select only one output, iteratively, per block. This avoids having tons of tiny tiny outputs.

### CODE REVIEW

File: src/minerfund.cpp [hunk 1]
```diff
@@ -43,6 +47,16 @@ std::vector<CTxOut> GetMinerFundRequiredOutputs(const Consensus::Params &params,
         return {};
     }
 
+    if (IsNumbersEnabled(params, pindexPrev)) {
+        const std::vector<std::string> &addresses =
+            GetPayoutAddresses(params, pindexPrev);
+        const size_t numAddresses = addresses.size();
+        const Amount shareAmount = blockReward / int64_t(2);
+        const auto addressIndx = numAddresses % pindexPrev->GetMedianTimePast();
```
Comment (schancel) [inline src/minerfund.cpp:55]: What was I smoking here. Intellisense...

File: src/consensus/addresses.h [hunk 2]
```diff
@@ -88,5 +88,34 @@ namespace RewardAddresses {
             "lotus_16PSJHuUsbDo5qZUjygQgtJLMsZgcpSB6wD459F6J",
             // Lotus Citizen 4
             "lotus_16PSJNEqzEoLhBigDN8Hf93FzERrPQKJXaYdzRCNT",
-        }};
+        },
+        .numbers = {
+            // Aleph 0
```
Comment (schancel) [inline src/consensus/addresses.h:93]: Needs to be updated.

**Review by maff1989 — APPROVED** [review]

**Review by MagsMagnoli — APPROVED** [review]

**Comment by nghiacc** [comment]
Confirmed for lotus_16PSJLMtCuFnyMb3rz4zbbhAtJgppp8GiY5NpGGev

**Comment by kensauruss** [comment]
Confirmed

**Pull request merged**